pkg/netdevice/netResourcePool.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netdevice
+
+import (
+	"github.com/golang/glog"
+	"github.com/intel/sriov-network-device-plugin/pkg/resources"
+	"github.com/intel/sriov-network-device-plugin/pkg/types"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+)
+
+type netResourcePool struct {
+	*resources.ResourcePoolImpl
+}
+
+var _ types.ResourcePool = &netResourcePool{}
+
+// NewNetResourcePool returns an instance of resourcePool
+func NewNetResourcePool(rc *types.ResourceConfig, apiDevices map[string]*pluginapi.Device, devicePool map[string]types.PciDevice) types.ResourcePool {
+	rp := resources.NewResourcePool(rc, apiDevices, devicePool)
+	return &netResourcePool{
+		ResourcePoolImpl: rp,
+	}
+}
+
+// Overrides GetDeviceSpecs
+func (rp *netResourcePool) GetDeviceSpecs(deviceIDs []string) []*pluginapi.DeviceSpec {
```

Comment:
This is a good way of also allowing Pool-Specific logic so it would help solve the problems described in #212 
However, only runtime logic is enabled. If discovery-time logic is desired, the logic would have to move to `NewNetResourcePool` and it would imply rewriting the already populated `deviceSpecs`, `mount` and `env`. So, I think that the separation of PoolDevice and PciDevice proposed in #213 would still be useful.

Reply:
The discovery logic per device/drivers is handled with "DeviceInfoProvider" interface, that we implemented for uio and vfio devices. The same interface could be implemented for other device use cases as well (e.g. vDPA devices, which is still a netdevice but needs additional information to be sent). So, if we could determine whether from sysfs or from the driver that this device will support vDPA then we could have vDPA specifc "DeviceInfoProvider" that will handle adding additional device files for DeviceSpecs.

Reply:
Unfortunately, AFAIK there is no way of getting that information from sysfs or the driver in a vendor-agnostic way for now.